src/axom/core/memory_management.hpp

```diff
@@ -197,6 +197,24 @@ inline void copy(void* dst, const void* src, std::size_t numbytes) noexcept;
 /// @}
 // _memory_management_routines_end
 
+/*!
+ * \brief Wrapper type representing an Umpire allocator ID.
+ *
+ *  This type is intended for use in function and constructor arguments, in
+ *  order to avoid ambiguities in overload resolution.
+ */
+struct Allocator
+{
+public:
+  explicit Allocator(int alloc_id = axom::getDefaultAllocatorID()) : m_id {alloc_id} { }
+
+  /// \brief Returns the allocator ID.
+  int get() const { return m_id; }
+
+private:
+  int m_id {0};
```

Comment:
Should this be `0` by default?  Or perhaps  `axom::INVALID_ALLOCATOR_ID`  
(which evaluates to `-1` and is not a valid umpire allocator ID).

Reply:
The default value of `axom::getDefaultAllocatorID()` is passed through the constructor, so I just removed the inline initializer.